app/Http/Controllers/API/AuthController.php

```diff
@@ -4,62 +4,66 @@
 
 use Illuminate\Http\Request;
 use App\Http\Controllers\API\ResponseController as ResponseController;
+use Illuminate\Support\Facades\Hash;
 use Illuminate\Support\Facades\Validator;
 use Illuminate\Support\Facades\Auth;
 use App\Models\User;
 
 class AuthController extends ResponseController
 {
     //create user
-    public function signup(Request $request)
-    {
+    public function signup(Request $request) {
         $validator = Validator::make($request->all(), [
-            'username' => 'required|string|',
-            'name' => 'required|string|',
-            'email' => 'required|string|email|unique:users',
-            'password' => 'required',
+            'username'         => 'required|string|unique:users',
+            'name'             => 'required|string|',
+            'email'            => 'required|string|email|unique:users',
+            'password'         => 'required',
             'confirm_password' => 'required|same:password'
```

Comment:
@HerrLevin Can you explain, why it is necessary to confirm the password via API? I understand such a procedure in the frontend, but in the API?

Reply:
It's stupid, you're right.